src/main/java/com/iota/iri/network/pipeline/TransactionProcessingPipelineImpl.java

```diff
@@ -52,7 +53,16 @@
 public class TransactionProcessingPipelineImpl implements TransactionProcessingPipeline {
 
     private static final Logger log = LoggerFactory.getLogger(TransactionProcessingPipelineImpl.class);
-    private ExecutorService stagesThreadPool = Executors.newFixedThreadPool(6);
+    private ExecutorService stagesThreadPool = Executors.newFixedThreadPool(getNumberOfThreads());
+
+    /**
+     * List of stages that will be ignored when determining thread count
+     */
+    private static List<Stage> IGNORED_STAGES = new LinkedList<Stage>(){{
```

Comment:
Change to 
    private static *final* List<Stage> IGNORED_STAGES = IotaUtils.createImmutableList ...

You can optionally refactor `createImmutableList` to `ImmutableList` in a separate commit (only if you feel the name is too long, else you can leave it)

Reply:
@DyrellC ?